### PR TITLE
theme & css stylekit (#572 #624)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
stylekit is commanded by the stylekit bool run parameter (default: True)
configurable through optional stylekit_config run parameter
`Gui(page=page).run(stylekit=True, stylekit_variables={"color_primary": "#00FF00"})`

### CODE REVIEW

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Excellent starting point.

A few changes that I think are relevant:

- Globally rename 'toolkit' to 'stylekit', rename 'toolkit_config' to 'stylekit_variables'.
     => git mv public/toolkit public/stylekit
- stylekit=True  should be the default, That should be the future of Taipy, with as little code as possible (no stylekit=True in every Taipy GUI app).
- homogeneity: please use the same coding style for css (space before {, endling line skip, 2 or 4 space chars for indentation, etc.)
- CSS file headers to be similar as well - including license?
-  There are very generic class names everywhere... like 'container'

**Comment by ghost** [comment]
@FredLL-Avaiga I think that's a great starting point indeed!
There is only one thing I would do differently, and it would be to move some theme styles from this `gui/src/themes/stylekit.ts` file directly to the relevant component's files, as styled component rules. This would be better for component specific styles to stay where they are useful rather than in a deported global file.
I can do that part if you like, in this branch or in a later one after you merge this.

@FabienLelaquais as for generic class names : "container", for example, is a name commonly used in multiple big and well known frameworks: CSS frameworks like Bootstrap or JS frameworks like Material UI (to cite just them) use this class (or component) name for that very specific case of limiting content width and centering it into the window. I get why you feel it's generic, but in the CSS/JS ecosystem, container (without affix) actually is a meaningful name.

**Review by ghost — APPROVED** [review]

**Comment by FredLL-Avaiga** [comment]
> move some theme styles from this `gui/src/themes/stylekit.ts` file directly to the relevant component's files, as styled component rules.

Why not but this need to be activated only if window.taipyConfig.stylekit ids defined ...

**Comment by ghost** [comment]
> this need to be activated only if window.taipyConfig.stylekit ids defined ...

Yes, but there are some normalization and improvement styles here that could be implemented right away, whether stylekit is ON or not, I think. Like making buttons and inputs the same height, or matching text inputs and selector styles, for obvious example.

File: gui/src/themes/stylekit.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+                root: {
+                    // Fill the available width
+                    display: "flex",
+
+                    "&.taipy-selector": {
+                        marginLeft: 0,
+                        marginRight: 0,
+                    },
+
+                    ".taipy-tree &": {
```
Comment (FredLL-Avaiga) [inline gui/src/themes/stylekit.ts]: @JaufreyTpy I don't understand these rules: for taipy-tree and taipy-layout, the components are not embedded in a FormControl ?
(this is valid for taipy-selector though and indeed I will apply it directly to the taipy component)

Comment (FredLL-Avaiga) [reply gui/src/themes/stylekit.ts]: oh sorry, I think I read it wrong.

Comment (FredLL-Avaiga) [reply gui/src/themes/stylekit.ts]: for taipy-tree and taipy-layout, these rules address the included FormControl (ie the TextField inside the taipy-tree Box)

**Comment by FabienLelaquais** [comment]
The last commit looks neat to me: a straightforward way to get tests through.
Now... would it be possible before the merge to "git mv gui/public/stylekit/visual-elements/taipy-blocks" and "taipy-controls" to "blocks" and "controls" respectively? That prefix (even worse at this level) makes very little sense to me
Thx

File: src/taipy/gui/gui.py [hunk 2]
```diff
@@ -244,8 +244,9 @@ def __init__(
         self._flask = flask
         if css_file is None:
             script_file = pathlib.Path(self.__frame.f_code.co_filename or ".").resolve()
-            css_file = f'{script_file.stem or "Taipy"}.css'
-        self._css_file = css_file
+            if script_file.with_suffix(".css").exists():
```
Comment (FabienLelaquais) [inline src/taipy/gui/gui.py:247]: Finally! Thx!

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**